src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -186,12 +174,7 @@ public static unsafe Array CreateInstance(Type elementType, int[] lengths, int[]
         [RequiresDynamicCode("The code for an array of the specified type might not be available.")]
         public static Array CreateInstance(Type elementType, params long[] lengths)
         {
-            if (lengths == null)
-            {
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.lengths);
-            }
-            if (lengths.Length == 0)
-                ThrowHelper.ThrowArgumentException(ExceptionResource.Arg_NeedAtLeast1Rank);
```

Comment:
Where exactly is the `lengths.Length == 0` check performed now?

Reply:
It will be performed by the `int[]` overload that this forwards to.